src/components/Select.vue

```diff
@@ -850,8 +867,13 @@ export default {
         openIndicator: {
           attributes: {
             ref: 'openIndicator',
-            role: 'presentation',
             class: 'vs__open-indicator',
+            type: 'button',
+            tabindex: -1, // Remove from tab sequence as tabbing to the input combobox opens the dropdown
+            'aria-hidden': 'true',
+          },
+          events: {
+            mousedown: this.toggleDropdown,
```

Comment:
And shouldn't there be a click instead of a mousedown event on a button?

Reply:
> For example, `type: 'button'` means, `#open-indicartor` is supposed to be a button.
> 
> But then why is it hidden?

The type=button is just for correct HTML even though it is not to be detected by screen readers as this is not a submit button

We want to only have one trigger for the dropdown which is keyboard focus on the input so we hide it from the tab sequence

> If it is a button to control the combobox, it should be visible and have `aria-expanded` and `aria-controls`. Only the icon should be hidden.

As the aria attrs are used by screen readers for keyboard users these would be extraneous as they would never be used due to being hidden

> And shouldn't there be a click instead of a mousedown event on a button?

This is to prevent a conflict where a single click triggers a rapid doulble toggle of the dropdown

Reply:
> The type=button is just for correct HTML even though it is not to be detected by screen readers as this is not a submit button

But it is not correct now. Those attributes are set on the span with an icon.

Reply:
> As the aria attrs are used by screen readers for keyboard users

Screen reader also works by mouse pointer and its own custom keyboard navigation.

Anyway, this is a part of the pattern on w3c from the issue...

Reply:
> they would never be used due to being hidden

But the button is not hidden. It is a visible and working button

Reply:
> But it is not correct now. Those attributes are set on the span with an icon, not the button.

Yes, will move them it to the `button` element here and custom `button` passed in the slot will have to add type=button itself

> Screen reader also works by mouse pointer and its own custom keyboard navigation.
> 
> Anyway, this is a part of the pattern on w3c from the issue...

In my testing with NVDA mouse hover at least does not work (see https://ux.stackexchange.com/questions/147234/do-accessibility-standards-wcag-aa-require-a-screen-reader-to-announce-an-html)

> But the button is not hidden. It is a visible and working button

Meant hidden from accessibility tree i.e. the screen reader

Reply:
Example of the issues in both this and prev comment:

![image](https://github.com/nextcloud-deps/vue-select/assets/25978914/44d699ee-a816-4e15-9276-3ee210a3148c)

1. Invalid HTML: `span` has `type="button"`
2. There is no new `Button`, because icon is passed to the icon slot, and it redefines a whole slot

Reply:
When the button will be here, it should not be hidden, a user can press on it, it is not decorative. (Only icon is hidden).

Reply:
> When the button will be here, it should not be hidden, a user can press on it, it is not decorative. (Only icon is hidden).

Yes, the button is only meant to be clicked on by mouse but not keyboard focused

Reply:
That explains hiding from keyboard access by `tabindex="-1"`, but not hiding from a screen reader by `aria-hidden`

Reply:
Does the screen reader announce it correctly for you with NVDA + mouse without aria-hidden? As due to the issue described in https://ux.stackexchange.com/questions/147234/do-accessibility-standards-wcag-aa-require-a-screen-reader-to-announce-an-html (also linked above) it does not announce it so it seems redundant

I want to keep aria-hidden to ensure it is removed from the accessibility tree completely as directly focusing on the toggle button via voice (have not tested) is made redundant by being able to focus the input

This also simplifies the UX I think for screen reader users when only controlling the input is needed to both toggle the dropdown and enter text

Reply:
> Does the screen reader announce it correctly for you with NVDA

On the pattern page - yes, on click it announces the button and its state aka `states button collapsed`. But not on hover.

> I want to keep aria-hidden to ensure it is removed from the accessibility tree completely as directly focusing on the toggle button via voice (have not tested) is made redundant by being able to focus the input

But screen readers are not only about voice access but also about reading the screen. It is still an existed button that can be clicked and used and a part of the UI (even if it is not required to use it)

Reply:
In the pattern from the issue, this button is not hidden and does have many `aria-*` attributes (that don't make sense for a hidden button).

https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/

I think, even if one or more screen readers don't announce the button on hover, we should still have all possibly useful attributes on it.

If a screen reader doesn't support it - these attributes don't hurt anyway, do they?

but if a screen reader support it - better to have all necessary attributes.